dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageProblem.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.ListMultimap;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Multimaps;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * A linkage error.
+ *
+ * @see <a href="https://jlbp.dev/glossary.html#linkage-error">Java Dependency Glossary: Linkage
+ *     Error</a>
+ */
+public final class LinkageProblem {
+
+  private final ErrorType errorType;
```

Comment:
In general "type" variables in classes are a code smell in object oriented languages. By default I expect types to be indicated by class/object types. Should that be done here?

Reply:
I'm not suggesting a renaming. I'm suggesting you don't wan this field at all. Type information should be conveyed by subtypes. Among other benefits that enables different subtypes to have methods and data specific to that type.

Reply:
So far, I don't see any need to have different subtypes. Current ErrorType enum is sufficient. When we encounter the need of differentiating them with error-specific fields, I'll apply the subtyping.